x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/rules/CombineDisjunctionsToInTests.java

```diff
@@ -129,4 +135,32 @@ public void testOrWithNonCombinableExpressions() {
         assertEquals(fa, in.value());
         assertThat(in.list(), contains(ONE, THREE));
     }
+
+    public void testCombineCIDRMatch() {
+        FieldAttribute faa = getFieldAttribute("a");
+        FieldAttribute fab = getFieldAttribute("b");
+
+        List<Expression> ipa1 = randomList(1, 5, () -> new Literal(EMPTY, randomLiteral(DataType.IP).value(), DataType.IP));
+        List<Expression> ipa2 = randomList(1, 5, () -> new Literal(EMPTY, randomLiteral(DataType.IP).value(), DataType.IP));
+        List<Expression> ipb1 = randomList(1, 5, () -> new Literal(EMPTY, randomLiteral(DataType.IP).value(), DataType.IP));
+        List<Expression> ipb2 = randomList(1, 5, () -> new Literal(EMPTY, randomLiteral(DataType.IP).value(), DataType.IP));
+
+        List<Expression> ipa = new HashSet<>(CollectionUtils.combine(ipa1, ipa2)).stream().toList();
+        List<Expression> ipb = new HashSet<>(CollectionUtils.combine(ipb1, ipb2)).stream().toList();
+
+        Or or1 = new Or(EMPTY, new CIDRMatch(EMPTY, faa, ipa1), new CIDRMatch(EMPTY, faa, ipa2));
+        Or or2 = new Or(EMPTY, new CIDRMatch(EMPTY, fab, ipb1), new CIDRMatch(EMPTY, fab, ipb2));
```

Comment:
You could maybe shuffle the `or`s creation, to each contain both `faa` and `fab`.

Reply:
Done.